src/lib/es5.d.ts

```diff
@@ -429,8 +429,8 @@ interface String {
 
     /**
      * Replaces text in a string, using a regular expression or search string.
-     * @param searchValue A string to search for.
-     * @param replaceValue A string containing the text to replace for every successful match of searchValue in this string.
+     * @param searchValue A string or regular expression to search for.
+     * @param replaceValue A string containing the text to replace. When the {@linkcode searchValue} is a RegExp, all matches are replaced if the g flag is set (taking into account the y flag if present). Otherwise {@linkcode searchValue} only the first match is replaced.

```

Comment:
```suggestion
     * @param replaceValue A string containing the text to replace. When the {@linkcode searchValue} is a `RegExp`, all matches are replaced if the `g` flag is set (or only those matches at the beginning, if the `y` flag if present). Otherwise, only the first match of {@linkcode searchValue} is replaced.
```

Nit

Reply:
I like this wording better, but note that the bit before the close paren should swap if->is: `flag is present` not `flag if present`.

Reply:
Ah yes—let's say "is also".